runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -280,135 +182,311 @@ public void onConfigChanged(ConfigChanged event)
 			{
-				checkExit();
+				frame.getRootPane().setWindowDecorationStyle(JRootPane.FRAME);
+
+				final JComponent titleBar = SubstanceCoreUtilities.getTitlePaneComponent(frame);
+				titleToolbar.putClientProperty(SubstanceTitlePaneUtilities.EXTRA_COMPONENT_KIND, SubstanceTitlePaneUtilities.ExtraComponentKind.TRAILING);
+				titleBar.add(titleToolbar);
+
+				// Substance's default layout manager for the title bar only lays out substance's components
+				// This wraps the default manager and lays out the TitleToolbar as well.
+				LayoutManager delegate = titleBar.getLayout();
+				titleBar.setLayout(new LayoutManager()
+				{
+					@Override
+					public void addLayoutComponent(String name, Component comp)
+					{
+						delegate.addLayoutComponent(name, comp);
+					}
+
+					@Override
+					public void removeLayoutComponent(Component comp)
+					{
+						delegate.removeLayoutComponent(comp);
+					}
+
+					@Override
+					public Dimension preferredLayoutSize(Container parent)
+					{
+						return delegate.preferredLayoutSize(parent);
+					}
+
+					@Override
+					public Dimension minimumLayoutSize(Container parent)
+					{
+						return delegate.minimumLayoutSize(parent);
+					}
+
+					@Override
+					public void layoutContainer(Container parent)
+					{
+						delegate.layoutContainer(parent);
+						final int width = titleToolbar.getPreferredSize().width;
+						titleToolbar.setBounds(titleBar.getWidth() - 75 - width, 0, width, titleBar.getHeight());
+					}
+				});
 			}
+
+			frame.pack();
+			SwingUtil.revalidateMinimumSize(frame);
+			frame.setLocationRelativeTo(frame.getOwner());
+			frame.setVisible(true);
+			frame.toFront();
+			requestFocus();
+			giveClientFocus();
 		});
 
-		final JPanel container = new JPanel();
-		container.setLayout(new BoxLayout(container, BoxLayout.X_AXIS));
-		container.add(new ClientPanel(client));
+		eventBus.post(new ClientUILoaded());
+	}
 
-		navContainer = new JPanel();
-		navContainer.setLayout(new BorderLayout(0, 0));
-		navContainer.setMinimumSize(new Dimension(0, 0));
-		navContainer.setMaximumSize(new Dimension(0, Integer.MAX_VALUE));
-		container.add(navContainer);
+	/**
+	 * Paint this component to target graphics
+	 *
+	 * @param graphics the graphics
+	 */
+	public void paint(final Graphics graphics)
+	{
+		frame.paint(graphics);
+	}
 
-		pluginToolbar = new PluginToolbar(this);
-		container.add(pluginToolbar);
+	/**
+	 * Gets component width.
+	 *
+	 * @return the width
+	 */
+	public int getWidth()
+	{
+		return frame.getWidth();
+	}
 
-		titleToolbar = new TitleToolbar(properties);
+	/**
+	 * Gets component height.
+	 *
+	 * @return the height
+	 */
+	public int getHeight()
+	{
+		return frame.getHeight();
+	}
 
-		add(container);
+	/**
+	 * Returns true if this component has focus.
+	 *
+	 * @return true if component has focus
+	 */
+	public boolean isFocused()
+	{
+		return frame.isFocused();
 	}
 
-	@Override
+	/**
+	 * Request focus on this component.
+	 */
```

Comment:
This function requests focus on the game applet, not this component

Reply:
It does both

Reply:
Only one can have focus, so it gets the frame to be focused, then immediately gives the game focus.

Reply:
Updated the description to state that first component requests focus and then client inside it.